### ISSUE TITLE
RichProgressBar in PyTorch Lightning only show progress at the very end

### ISSUE THREAD

**Issue by npuichigo** [ray-project/ray#42091]
### What happened + What you expected to happen

RichProgressBar in PyTorch Lightning only show progress at the very end

### Versions / Dependencies

2.9

### Reproduction script

```python
import torch
import lightning.pytorch as pl
from datasets import load_dataset

from transformers import (
    AutoModelForSequenceClassification,
    AutoTokenizer,
)

from ray.train.torch import TorchTrainer
from ray.train import DataConfig, ScalingConfig
import ray.train.lightning
from lightning.pytorch.callbacks import RichProgressBar


# Model, Loss, Optimizer
class SequenceClassifier(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.model = AutoModelForSequenceClassification.from_pretrained(
            "bert-base-cased", return_dict=True
        )

    def forward(self, x):
        return self.model(x)

    def training_step(self, batch, batch_idx):
        outputs = self.model(**batch)
        loss = outputs.loss
        self.log("loss", loss, on_step=True, prog_bar=True)
        return loss

    def configure_optimizers(self):
        return torch.optim.Adam(self.model.parameters(), lr=2e-5)


def train_func(config):
    train_ds = ray.train.get_dataset_shard("train")
    eval_ds = ray.train.get_dataset_shard("validation")

    tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")

    def collate_fn(batch):
        outputs = tokenizer(
            list(batch["sentence1"]),
            list(batch["sentence2"]),
            truncation=True,
            padding="longest",
            return_tensors="pt",
        )
        outputs["labels"] = torch.LongTensor(batch["label"])
        return outputs

    train_dataloader = train_ds.iter_torch_batches(
        batch_size=config["train_batch_size"], collate_fn=collate_fn
    )
    eval_dataloader = eval_ds.iter_torch_batches(
        batch_size=config["eval_batch_size"], collate_fn=collate_fn
    )
    model = SequenceClassifier()

    trainer = pl.Trainer(
        max_epochs=config["num_epochs"],
        devices="auto",
        accelerator="auto",
        strategy=ray.train.lightning.RayDDPStrategy(),
        plugins=[ray.train.lightning.RayLightningEnvironment()],
        callbacks=[ray.train.lightning.RayTrainReportCallback(), RichProgressBar()],
    )
    trainer = ray.train.lightning.prepare_trainer(trainer)
    trainer.fit(model, train_dataloaders=train_dataloader)


if __name__ == "__main__":
    config = {
        "num_epochs": 5,
        "train_batch_size": 32,
        "eval_batch_size": 32,
    }

    # Prepare Ray Datasets
    hf_datasets = load_dataset("glue", "mrpc")
    ray_datasets = {
        "train": ray.data.from_huggingface(hf_datasets["train"]),
        "validation": ray.data.from_huggingface(hf_datasets["validation"]),
    }
    config["train_dataset_size"] = ray_datasets["train"].count()

    trainer = TorchTrainer(
        train_func,
        train_loop_config=config,
        datasets=ray_datasets,
        dataset_config=DataConfig(datasets_to_split=["train", "validation"]),
        scaling_config=ScalingConfig(num_workers=1, use_gpu=True),
    )

    result = trainer.fit()
```

### Issue Severity

None

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
